x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -726,6 +784,30 @@ export class AlertsClient {
     }
   }
 
+  public async bulkUpdateCases({ ids, index, caseIds }: BulkUpdateCasesOptions) {
+    /**
+     * First check in case the caseIds are more that the allowed limit.
+     * We do this check to avoid any mget calls or authorization checks
+     *
+     * The check below does not ensure that an alert may exceed the limit
+     * if we add the casedIds. We need to also throw in case
+     * alert.caseIds + caseIds > MAX_CASES_PER_ALERT. The validateTotalCasesPerAlert
+     * function ensures that.
+     *
+     */
+    if (caseIds.length > MAX_CASES_PER_ALERT) {
+      throw Boom.badRequest(`You cannot attach more than ${MAX_CASES_PER_ALERT} cases to an alert`);
+    }
+
+    return this.mgetAlertsAuditOperate({
+      ids,
+      indexName: index,
+      operation: ReadOperations.Get,
```

Comment:
A user with `read` access to an alert and `write` access to a case should be able to link the case to the alert (update the alert's data to include the case ids). For that reason, the operation is a read operation.